### ISSUE TITLE
Support Enums with Generic Type Parameters

### ISSUE THREAD

**Issue by eignnx** [cyqsimon/documented#9]
I love the idea for this crate! Unfortunately when I tried to use it I ran into some trouble because the enum I wanted to apply `DocumentedVariants` to has a generic type parameter `L`:

```rust
#[derive(.., Clone, .., documented::DocumentedVariants)]
pub enum Instr<L> {
    /// Dispatch on the type of the value pointed to by `X1`.
    SwitchOnTerm { .. }
    // ..
}
```

The error I get says:

```
error[E0107]: missing generics for enum `Instr`
  --> src\bc\instr.rs:93:10
   |
93 | pub enum Instr<L> {
   |          ^^^^^ expected 1 generic argument
   |
note: enum defined here, with 1 generic parameter: `L`
  --> src\bc\instr.rs:93:10
   |
93 | pub enum Instr<L> {
   |          ^^^^^ -
help: add missing generic argument
   |
93 | pub enum Instr<L><L> {
   |               +++

```

I figured out a workaround though.

1. In my case I could assign the type `()` as a default for `L`:
   
    ```rust
    pub enum Instr<L = ()> {
    ```
2. Then I wrote a helper impl:
    ```rust
    impl<L: Clone> Instr<L> {
        pub fn doc_comment(&self) -> Option<&'static str> {
            let mapped: Instr<()> = self.clone().map_lbl(|_| ()); // Throw away `L`, make it `()`.
            documented::DocumentedVariants::get_variant_docs(&mapped).ok()
        }
    }
    ```
    Unfortunately this requires that `L` be `Clone` (still ok for my use case, but unsatisfying).

    Here `map_lbl` maps the `L` parameter to a new type. It's signature looks like this:
    ```rust
    impl<L> Instr<L> {
        pub fn map_lbl<M>(self, f: impl Fn(L) -> M) -> Instr<M> { .. }
    }
    ```

I'm not sure how the derive macro works, but it seems like handling generic type parameters just hasn't been implemented yet. Would this be easy to fix?

**Comment by cyqsimon** [comment]
Ah yeah generics was something I didn't consider, primarily because I personally didn't need it.

It's just an extra case that needs to be handled, so it shouldn't be too much work.

**Comment by cyqsimon** [comment]
Fix released in 0.4.3.

0.4.2 was yanked because my sleep-deprived brain borked dependency versions. Time to get 10 hours of sleep I guess.
